README.md

```diff
@@ -1,33 +1,131 @@
 # obsForge
 Forging next generation of observation processing leveraging JEDI
 
-# Clone and Build
-```
+## Clone and Build
+```bash
 git clone --recursive --jobs 2 https://github.com/NOAA-EMC/obsForge.git
 cd obsForge
 ./build.sh
 ```
 
-# Tests
-Load the modules if you have not yet,
+## Testing
+
+This section provides instructions for running all tests in the main obsForge repository (excluding submodule tests). The testing includes both C++ tests managed by CTest and Python tests using pytest.
+
+### Prerequisites
+
+1. **Build the project** (if not already done):
+   ```bash
+   ./build.sh
+   ```
+
+2. **Load modules** (on supported HPC systems):
+   ```bash
+   module use modulefiles
+   module load obsforge/{hpc}.{compiler}
+   ```
+
+3. **Set up Python environment** for pytest:
+   ```bash
+   python3 -m venv obsforge_test_env
+   source obsforge_test_env/bin/activate
+   pip install -r requirements.txt
+   ```
```

Comment:
I don't remember if this is necessary after loading the modules. It's definitely needed when using the JCSDA docker container.